src/nsolid.h

```diff
@@ -1015,6 +1015,33 @@ class NODE_EXTERN Snapshot {
    */
   static int StopTrackingHeapObjectsSync(SharedEnvInst envinst);
 
+  /**
+   * @brief Will start sampling heap allocations.
+   *
+   * @param envinst SharedEnvInst of thread to take the snapshot from.
+   * @param duration duration in milliseconds of the heap profiler after which
+   * the heap snapshot will be returned in the callback.
+   * @param cb callback function with the following signature:
+   * `cb(int status, std::string snapshot, ...Data)`. It will be called from the
+   * NSolid thread.
+   * @param data variable number of arguments to be propagated to the callback.
+   * @return NSOLID_E_SUCCESS in case of success or a different NSOLID_E_
+   * error value otherwise.
+   */
+  template <typename Cb, typename... Data>
+  static int StartSampling(SharedEnvInst envinst,
+                           uint64_t duration,
+                           Cb&& cb,
+                           Data&&... data);
+
```

Comment:
How do I configure the other parameters allowed for sampling. I propose smthg like this:
```c++
template <typename Cb, typename... Data>
  static int StartSampling(SharedEnvInst envinst,
                           uint64_t duration,
                           Cb&& cb,
                           Data&&... data);
  template <typename Cb, typename... Data>
  static int StartSampling(SharedEnvInst envinst,
                           uint64_t sample_interval,
                           int stack_depth,
                           v8::HeapProfiler::SamplingFlags flags,
                           uint64_t duration,
                           Cb&& cb,
                           Data&&... data);
```
where the former calls with the default values to the latter

Reply:
Also, does redacting the sampling data apply here?

Reply:
> Also, does redacting the sampling data apply here?

Nope, this does not include any sensitive information.